### PR TITLE
Creating Locale extension to determine language layout direction

### PR DESCRIPTION
**Pull request by sophialee0416**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Moving isLeftToRight() method from AvatarGroup (#662) to a shared location within Core, and making it an extension.

### Verification

Tested RTL languages, changes remained consistent with that of #662.

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] VoiceOver and Keyboard Accessibility
- [x] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/663)

### CODE REVIEW

File: ios/FluentUI/Core/SwiftUI+ViewModifiers.swift [hunk 1]
```diff
@@ -61,3 +61,16 @@ extension View {
                               shouldScale: shouldScale))
     }
 }
+
+extension Locale {
```
Comment (harrieshin) [inline ios/FluentUI/Core/SwiftUI+ViewModifiers.swift:65]: hmmm it is nothing to do with the view modifier. might as well create a new file?

File: ios/FluentUI/Core/SwiftUI+ViewModifiers.swift [hunk 2]
```diff
@@ -61,3 +61,16 @@ extension View {
...
+    /// Determines if layout direction of current language is `.leftToRight`.
+    /// - Returns: True if layout direction is left to right, false otherwise.
+    func isLeftToRightLayoutDirection() -> Bool {
```
Comment (harrieshin) [inline ios/FluentUI/Core/SwiftUI+ViewModifiers.swift:68]: supernit: I can be swayed either direction(no pun intended) but there are more api that check RightToLeft than LeftToright in fluent.

File: ios/FluentUI/Core/SwiftUI+ViewModifiers.swift (hunk 1 above)
Comment (sophialee0416) [reply ios/FluentUI/Core/SwiftUI+ViewModifiers.swift:65]: Would you recommend creating a new file or renaming this one to just SwiftUI+Modifiers?

File: ios/FluentUI/Core/SwiftUI+ViewModifiers.swift (hunk 2 above)
Comment (sophialee0416) [reply ios/FluentUI/Core/SwiftUI+ViewModifiers.swift:68]: True, let me adjust!

File: ios/FluentUI/Core/SwiftUI+ViewModifiers.swift (hunk 1 above)
Comment (mischreiber) [reply ios/FluentUI/Core/SwiftUI+ViewModifiers.swift:65]: Personally I vote for a separate file - I'd rather a dozen files with a single method each, versus a single file with a dozen unrelated objects (which is what will probably happen to this file eventually if we make it not View-specific).

A new file should go in the Extensions folder with a name `Locale+Extensions.swift`

**Review by harrieshin — APPROVED** [review]

File: ios/FluentUI/Core/Locale+Extensions.swift [hunk 3]
```diff
@@ -0,0 +1,19 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
```
Comment (mischreiber) [inline ios/FluentUI/Core/Locale+Extensions.swift:6]: Looks good! Could you just move this to the `Extensions/` directory? 

https://github.com/sophialee0416/fluentui-apple/tree/sople/locale/ios/FluentUI/Extensions

**Review by mischreiber — CHANGES_REQUESTED** [review]

**Review by mischreiber — APPROVED** [review]

**Pull request merged**